File: campus-trade-user/src/api/report.ts
import request from '@/utils/request'
import type { PageResult } from '@/types'

export interface ReportVO {
  id: number
  reporterId: number
  reporterName: string
  targetType: number
  targetId: number
  reason: string
  description: string
  status: string
  handleResult: string
  createTime: string
}

export const createReport = (data: { targetType: number; targetId: number; reason: string; description?: string; images?: string }) =>
  request.post('/report', data)

export const listMyReports = (pageNum: number = 1, pageSize: number = 10) =>
  request.get<never, PageResult<ReportVO>>('/report/mine', { params: { pageNum, pageSize } })

File: campus-trade-server/src/main/java/com/campustrade/entity/Report.java
package com.campustrade.entity;

import com.campustrade.base.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = true)
public class Report extends BaseEntity {

    private Long reporterId;
    private Integer targetType;
    private Long targetId;
    private String reason;
    private String description;
    private String images;
    private String status;
    private Long handlerId;
    private String handleResult;
    private LocalDateTime handleTime;
}